ehitchhiking-ui/src/app/pages/main-screen/yandex-map/yandex-map.component.ts

```diff
@@ -1,24 +1,166 @@
+		if (changes.tripData && changes.tripData.currentValue) {
+			if (this.userState === UserState.driver) {
+				this.myMap.geoObjects.remove(this.currentMultiRoute);
+				console.log(this.tripData);
+				this.addMultiRoute(this.tripData, true);
+			}
+			if (this.userState === UserState.passenger) {
+				// фильтрация коллекции activeRoutesCollection
+				for (let i = 0; i < 3; i++) {
+					this.addMultiRoute(this.activeRoutesCollection[i], false);
+				}
+			}
+		}
+		if (changes.userState && changes.userState.currentValue) {
+			this.setUserIconToMapAccordingUserState();
+		}
+		if (changes.isSavedRoute && this.currentMultiRoute) {
+			this.activeRoutesCollection.push(this.tripData);
+			this.myMap.geoObjects.remove(this.currentMultiRoute);
+		}
+	}
+
+	createMap() {
+		this.ymapsPromise
+			.then((maps) => {
+				this.myMap = new maps.Map(
+					'map',
+					{
+						center: [53.9, 27.5666], // Minsk coordinates
+						zoom: 10,
+					},
+					{searchControlProvider: 'yandex#search'}
+				);
+				maps.geolocation
+					.get({
+						mapStateAutoApply: true,
+					})
+					.then((result) => {
+						this.currentGeoPosition = result.geoObjects;
+						console.log(result.geoObjects); //
+						result.geoObjects.options.set('preset', 'islands#redPersonCircleIcon');
+						this.myMap.geoObjects.add(this.currentGeoPosition);
+					});
+				new maps.SuggestView('suggest1');
+				new maps.SuggestView('suggest2');
+			})
+			.catch((error) => console.log('Failed to load Yandex Maps', error));
+	}
+
+	addMultiRoute(data: Partial<Route>, pointDraggable: boolean) {
+		const color = this.colors[this.getRandomInt(0, 10)];
+		data.routeColor = color;
+		this.ymapsPromise
 			.then((maps) => {
-				const map = new maps.Map('map', {
-					center: [-8.369326, 115.166023],
-					zoom: 7,
-				});
+				const multiRoute = new maps.multiRouter.MultiRoute(
+					{
+						referencePoints: [data.from, data.to],
+						params: {
+							results: 1,
+						},
+					},
+					YandexMapService.routeOptions(color, pointDraggable)
+				);
+				console.log(data);
+				this.setInfoAboutRoute(multiRoute, data);
+				this.myMap.geoObjects.add(multiRoute);
+				this.currentMultiRoute = multiRoute;
 			})
 			.catch((error) => console.log('Failed to load Yandex Maps', error));
 	}
+
+	setInfoAboutRoute(multiRoute, data: Partial<Route>) {
+		multiRoute.events.add('update', (e) => {
+			const activeRoute = multiRoute.getActiveRoute();
+			data.tripDuration = activeRoute.properties.get('duration').text;
+			activeRoute.events.add('mouseenter', () => {
+				activeRoute.options.set('strokeWidth', 5);
+				activeRoute.options.set('strokeColor', '#fb5b74');
+			});
+			activeRoute.events.add('mouseleave', () => {
+				activeRoute.options.unset('strokeColor');
+			});
+			activeRoute.events.add('click', (event) => {
+				if (!this.myMap.balloon.isOpen()) {
+					const coords = event.get('coords');
+					this.myMap.balloon.open(coords, YandexMapService.baloonInfo(data));
+				} else {
```

Comment:
else block is excess if there is return in if

Reply:
fixed